core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -383,6 +383,9 @@ private long catchUp(JournalReader journalReader) {
       if (entry == null) {
         return journalReader.getNextSequenceNumber();
       }
+      if (journalReader.isReadingNewCheckpoint()) {
```

Comment:
Could we determine this by the sequence number of the read journal entry? If the sequence number is `0`, we call reset.

Reply:
done